src/v/raft/consensus.cc

```diff
@@ -3832,4 +3832,26 @@ void consensus::upsert_recovery_state(
     }
 }
 
+ss::future<> consensus::maybe_flush_log(size_t threshold_bytes) {
+    // if there is nothing to do exit without grabbing an op_lock
+    // this condition will be validated again under the op_lock in `flush_log`
```

Comment:
Probably not a big deal, but flush_log doesn't validate exactly this condition, right? It just checks if there is _something_ to flush.

Reply:
updated comment